src/main/java/mate/academy/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,212 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateShoppingCartsStatement
+                        = connection.prepareStatement(updateShoppingCarts)) {
+            updateShoppingCartsStatement.setLong(1, id);
+            if (updateShoppingCartsStatement.executeUpdate() == 0) {
+                return false;
+            }
+            deleteShoppingCartProduct(id, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting shopping cart with id="
+                    + id + " is failed", e);
+        }
+        return true;
+    }
+
+    private void setProductId(Product product, Connection connection)
+            throws SQLException {
+        String selectProducts = "SELECT product_id FROM products "
+                + "WHERE product_name = ? AND deleted = false;";
+        try (PreparedStatement selectProductsStatement
+                     = connection.prepareStatement(selectProducts)) {
+            selectProductsStatement.setString(1, product.getName());
+            ResultSet resultSet = selectProductsStatement.executeQuery();
+            resultSet.next();
+            product.setId(resultSet.getLong(1));
+        }
+    }
+
+    private List<Product> getProducts(Long cartId, Connection connection)
+            throws SQLException {
+        String selectProduct = "SELECT products.product_id, product_name, product_price "
+                + "FROM shopping_carts_products "
+                + "JOIN products "
+                + "ON shopping_carts_products.product_id = products.product_id "
+                + "WHERE shopping_carts_products.cart_id = ?;";
+        List<Product> products = new ArrayList<>();
+        try (PreparedStatement selectProductStatement
+                     = connection.prepareStatement(selectProduct)) {
+            selectProductStatement.setLong(1, cartId);
+            ResultSet resultSet = selectProductStatement.executeQuery();
+            while (resultSet.next()) {
+                Long productId = resultSet.getLong(1);
+                String productName = resultSet.getString(2);
+                double productPrice = resultSet.getDouble(3);
+                Product product = new Product(productName, productPrice);
+                product.setId(productId);
+                products.add(product);
+            }
+        }
+        return products;
+    }
+
+    private void insertShoppingCartsProducts(ShoppingCart shoppingCart, Connection connection)
+            throws SQLException {
+        String insertShoppingCartsProducts
+                = "INSERT INTO shopping_carts_products (cart_id, product_id) VALUES (?, ?);";
+        try (PreparedStatement insertStatement
+                     = connection.prepareStatement(insertShoppingCartsProducts)) {
+            for (Product product : shoppingCart.getProducts()) {
+                setProductId(product, connection);
```

Comment:
in update method you should already have products with ids, no need to fetcher them additionally here